metrics-collectd/src/main/java/com/codahale/metrics/collectd/Collectd.java

```
@@ -0,0 +1,156 @@
+
+    public void disconnect() throws IOException {
+        if (channel == null) {
+            return;
+        }
+        try {
+            channel.close();
+        } finally {
+            channel = null;
+        }
+    }
+
+    public void send(Identifier id, long value, long timestamp, DataType dataType, long period) {
+        send(id, toByteArray(value), timestamp, dataType, period);
+    }
+
+    public void send(Identifier id, double value, long timestamp, DataType dataType, long period) {
+        send(id, toByteArray(value), timestamp, dataType, period);
+    }
+
+    private void send(Identifier identifier, byte[] value, long timestamp, DataType dataType, long period) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream(Network.BUFFER_SIZE);
+        DataOutputStream buffer = new DataOutputStream(baos);
+        try {
+            writeString(buffer, Network.TYPE_HOST, identifier.host);
+            writeNumber(buffer, Network.TYPE_TIME, timestamp / 1000);
+            writeString(buffer, Network.TYPE_PLUGIN, identifier.plugin);
+            writeString(buffer, Network.TYPE_PLUGIN_INSTANCE, identifier.pluginInstance);
+            writeString(buffer, Network.TYPE_TYPE, identifier.type);
+            writeString(buffer, Network.TYPE_TYPE_INSTANCE, identifier.typeInstance);
+            writeNumber(buffer, Network.TYPE_INTERVAL, period);
+
+            int len = Network.HEADER_LEN + Network.UINT16_LEN + Network.UINT8_LEN + Network.UINT64_LEN;
+            writeHeader(buffer, Network.TYPE_VALUES, len);
+            buffer.writeShort((short) 1);
+            buffer.write(dataType.getCode());
+        } catch (IOException e) {
+            LOG.error("unable to write bytearray", e);
+            return;
+        }
+        try {
+            buffer.write(value);
+            buffer.close();
+        } catch (IOException e) {
+            LOG.error("unable to write bytearray", e);
+        }
+        if (buffer.size() > Network.BUFFER_SIZE) {
```

If we use a `ByteBuffer` this check will become superfluous, because `ByteBuffer` can't grow and will throw an `IllegalArgumentException` if metric data exceeds its capacity. The biggest issue is that this exception will propagated instead of logged, but I think it's a good think, because I presume such failures can't be easily fixed.